workspaces/spectacle/src/index.ts

```diff
@@ -144,23 +144,14 @@ export interface IOpticContext {
 
 ////////////////////////////////////////////////////////////////////////////////
 
-export type AsyncStatus<T> =
-  | {
-      loading: true;
-      error: false;
-      data: null;
-    }
-  | {
-      loading: false;
-      error: true;
-      data: null;
-    }
-  | { loading: false; error: false; data: T };
-
 export interface IBaseSpectacle {
-  query(options: SpectacleInput): Promise<any>;
+  query<Result, Input = {}>(
```

Comment:
I feel like there should be a way to make these typescript generics reusable (since they are used in a bunch of places) but I'm not sure if this is possible / how to do this